src/main/java/org/commcare/data/xml/VirtualInstances.java

```diff
@@ -24,4 +28,15 @@ public static VirtualDataInstance buildSearchInputInstance(Map<String, String> u
         TreeElement root = TreeBuilder.buildTree(SEARCH_INSTANCE_ID, SEARCH_INSTANCE_ROOT_NAME, nodes);
         return new VirtualDataInstance(SEARCH_INSTANCE_ID, root);
     }
+
+    public static VirtualDataInstance buildSelectedValuesInstance(
+            ExternalDataInstance instance, String[] selectedValues) {
+        List<SimpleNode> nodes = new ArrayList<>();
+        for (String selectedValue : selectedValues) {
+            nodes.add(SimpleNode.textNode(SELCTED_CASES_INSTANCE_NODE_NAME, selectedValue));
+        }
+        TreeElement root = TreeBuilder.buildTree(instance.getInstanceId(), SELCTED_CASES_INSTANCE_ROOT_NAME,
+                nodes);
+        return new VirtualDataInstance(SEARCH_INSTANCE_ID, root);
```

Comment:
This needs a different instance ID
```suggestion
        return new VirtualDataInstance("search-input", root);
```

Reply:
thanks for catching, [Corrects instanceid](https://github.com/dimagi/commcare-core/pull/1084/commits/6ea2a14df19497d67239586b8992aee0fa4a6d39)